agents/kathy/src/settings.rs

```diff
@@ -11,15 +11,61 @@ mod test {
     use nomad_base::{get_remotes_from_env, NomadAgent};
     use nomad_test::test_utils;
     use nomad_xyz_configuration::{agent::kathy::ChatGenConfig, AgentSecrets};
+    use serde_json::json;
 
-    #[test]
+    #[tokio::test]
     #[serial_test::serial]
-    fn it_overrides_settings_from_env() {
-        test_utils::run_test_with_env_sync("../../fixtures/env.test-agents", move || {
+    async fn it_loads_remote_config() {
+        let config =
+            nomad_xyz_configuration::NomadConfig::from_file("../../fixtures/external_config.json")
+                .unwrap();
+        let config_json = json!(config).to_string();
+        test_utils::run_test_with_env_http(
+            "../../fixtures/env.external",
+            config_json,
+            |url| async move {
+                std::env::set_var("CONFIG_URL", url);
+
+                let agent_home = dotenv::var("AGENT_HOME_NAME").unwrap();
+
+                let settings = KathySettings::new().await.unwrap();
+
+                let remotes = get_remotes_from_env!(agent_home, config);
+                let mut networks = remotes.clone();
+                networks.insert(agent_home.clone());
+
+                let secrets = AgentSecrets::from_env(&networks).unwrap();
+                secrets
+                    .validate("kathy", &networks)
+                    .expect("!secrets validate");
+
+                settings
+                    .base
+                    .validate_against_config_and_secrets(
+                        crate::Kathy::AGENT_NAME,
+                        &agent_home,
+                        &remotes,
+                        &config,
+                        &secrets,
+                    )
+                    .unwrap();
+
+                let agent_config = &config.agent().get("ethereum").unwrap().kathy;
+                assert_eq!(settings.agent.interval, agent_config.interval);
+                assert_eq!(settings.agent.chat, agent_config.chat);
```

Comment:
I would add assert statement to ensure the agent has settings that matches `fixtures/external_config.json`, since that's what is served at `CONFIG_URL`. E.g. agent config only has two networks specified in `fixtures/external_config.json`